tasks/config_logstash.yml

```diff
@@ -50,6 +50,12 @@
     - /opt/logstash/lib/logstash/outputs/elasticsearch
     - /opt/logstash/patterns
 
+- name: config_logstash | add the logstash user to system groups
+  user:
+    name: "{{ logstash_system_user }}"
+    groups: "{{ logstash_system_groups | join(',') }}"
+    append: no
```

Comment:
If `logstash_system_groups: []` then this task will remove the `logstash_system_user` from all groups. Should probably at least include the group `logstash` in `logstash_system_groups`. We need to also update any hardcoded `user: logstash` tasks to reference `logstash_system_user` as well. The other option is to change `append: no` to `append: yes`.

Reply:
There's a separate option in the user module for specifying the main system group logstash is in called `group`.

`groups` only holds secondary groups. Setting it to a blank string by default removes the logstash user from any other groups apart from the `logstash` group.

I explicitly set `append: no` in an effort to make the task idempotent. This should be ok, since the main group is not affected at all.